lib/rules/no-cond-assign.js

```diff
@@ -106,7 +106,7 @@ module.exports = {
 
                 context.report({
                     node,
-                    loc: node.test.loc.start,
+                    loc: node.test.loc,
```

Comment:
I assume that if we are reporting the `node.test` location (`node.test.loc`), then `node.type === "IfStatement"`.

Why not just report `node: node.test` (on line 108)? Then the code automatically uses the location of that node (i.e., `node.test.loc`) when preparing the lint report.

I don't think we are getting any value in reporting on the IfStatement but setting the location equal to the AssignmentExpression's location.

Reply:
Ah, sorry I didn't understand the first time!

Since the rule targets the assignment, I completely agree. I'm not sure why was the `IfStatement` reported. Maybe because the message is `"Expected a conditional expression and instead saw an assignment."`. There is a following comment in the code:

```js
// must match JSHint's error message
missing: "Expected a conditional expression and instead saw an assignment."
```

Reply:
#4040 might be relevant to this question. The `loc` was added to point to the `test` node instead of just changing the reported node to the `test` node. I'm not sure why.

Reply:
The changes in this PR indeed look really unusual compared to all other rules.

Looks much better to simply report the `AssignmentExpression` node, as you suggested.

I'll modify this, which will change `nodeType` in the linting output, I guess that isn't information of particular importance?

Reply:
This is done now.